### ISSUE TITLE
Fix `unit-no-unknown` false positives for font-relative length units

### ISSUE THREAD

**Issue by ybiquitous** [stylelint/stylelint#6373]
### What steps are needed to reproduce the bug?

For example, the CSS code using the `ic` unit:

```css
a { width: 10ic; }
```

We can reproduce the bug on the [demo](https://stylelint.io/demo/#N4Igxg9gJgpiBcICGACYKDuBLKAXAFvCgIwAMWYA3CgL4A6AdiADQgDOAng7kgB4Lg2bFuAgMAZlgDmA4IxQo6IAE4BXADYw2SonIYKFS1Qyy4AtAwhnjAa0sYGOlLjUx5tRjRA0gA).

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "unit-no-unknown": true
  }
}
```

### How did you run Stylelint?

CLI with `stylelint *.css`

### Which version of Stylelint are you using?

14.13.0

### What did you expect to happen?

No problems are to be reported.

### What actually happened?

The following problem is reported:

```
test.css
 1:14  ✖  Unexpected unknown unit "ic"  unit-no-unknown

1 problem (1 error, 0 warnings)
```

### Does the bug relate to non-standard syntax?

No.

### Proposal to fix the bug

To Update the internal list for the units:

https://github.com/stylelint/stylelint/blob/eee9deb35190f46c5f964769d521fef5d7d1d7a8/lib/reference/units.js#L5-L12
